sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpecs.java

```diff
@@ -186,6 +243,16 @@ private ValueStateSpec(Coder<T> coder) {
     }
 
     @Override
+    public void offerCoders(Coder[] coders) {
+      if (this.coder == null) {
+        if (coders[0] != null) {
+          //noinspection unchecked
```

Comment:
Yea, we have a lot of things disabled that it would be nice to enable one module at a time until we can stop suppressing warnings. This one is actually pretty valuable to have enabled for code health IMO.